socks5.py

```diff
@@ -402,8 +508,37 @@ def run_wpad_server(server):
     except KeyboardInterrupt:
         pass
 
+def print_traffic_info():
+    global inbound_traffic
+    global outbound_traffic
+    while True:
+        time.sleep(30)
+        with traffic_lock:
+            inbound_mbps = (inbound_traffic * 8) / (30 * 1024 * 1024)
+            outbound_mbps = (outbound_traffic * 8) / (30 * 1024 * 1024)
+            inbound_traffic = 0
+            outbound_traffic = 0
+        # Clear the console
+        if 'Pythonista' in sys.executable:
+            console.clear()
+        else:
+            print('\033c', end='')
+# print initial_output
+        print(initial_output)
+        print("PAC URL: http://{}:{}/wpad.dat".format(PROXY_HOST, WPAD_PORT))
+        print("SOCKS Address: {}:{}".format(PROXY_HOST or SOCKS_HOST, SOCKS_PORT))
+
+        # Print the table
+        print(f"{'Direction':<10} | {'Traffic (Mbps)':<15}")
+        print(f"{'-'*10} | {'-'*15}")
+        print(f"{'Inbound':<10} | {inbound_mbps:<15.2f}")
+        print(f"{'Outbound':<10} | {outbound_mbps:<15.2f}")
 
 if __name__ == '__main__':
+    traffic_thread = threading.Thread(target=print_traffic_info)
+    traffic_thread.daemon = True
+    traffic_thread.start()
```

Comment:
Make sure to shut down your thread on exit (in the `KeyboardInterrupt` block). When you stop a script, all that happens is it receives a KeyboardInterrupt; threads that existed previously will continue running, which will rather badly mess up future relaunches of the script.

Reply:
Fixed in the latest commit